### PR TITLE
Update DG diagrams and add Storage instructions

### PR DESCRIPTION
**Pull request by muhammad-khair**
See: https://muhammad-khair.github.io/tp/DeveloperGuide.html

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1127,10 +1246,94 @@ testers are expected to do more *exploratory* testing.
...
+       <center>
+       <img src="images/DeveloperGuideImage/exit_ui.png"/>
+       </center><br>
+       Expected: The UI should display the elements correctly and the window should close.<br>
+
+#### Managing data externally
+
+1. Loading the application with no data files present
+
+    1. Prerequisites: There should be no files present in the directory you are going to work with, apart from `TuitiONE.jar`.
```
Comment (eezj35) [inline docs/DeveloperGuide.md:1285]: Should we use bullet point instead?

**Review by eezj35 — CHANGES_REQUESTED** [review]
just that minor part about using bullet points, but other than that thanks for the hard work 💯 

image

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (muhammad-khair) [reply docs/DeveloperGuide.md:1285]: I'm fine with that, but the rest need to standardise for their ongoing PRs as well

**Review by Timothyoung97 — APPROVED** [review]
LGTM

**Pull request merged**